server/routes/bulkOffchain.ts

```diff
@@ -15,266 +15,293 @@ import { RoleInstance } from '../models/role';
+          //       chain: t.addr_chain,
+          //     },
+          //   };
+          //   if (t.topic_id) {
+          //     data['topic'] = {
+          //       id: t.topic_id,
+          //       name: t.topic_name,
+          //       description: t.topic_description,
+          //       communityId: t.topic_community,
+          //       chainId: t.topic_chain,
+          //       telegram: t.telegram,
+          //     };
+          //   }
+          //   return data;
+          // });
+        } catch (e) {
+          console.log(e);
+          reject(new Error('Could not fetch threads, comments, or reactions'));
+        }
+      }),
+      // admins
+      models.Role.findAll({
+        where: chain
+          ? {
+              chain_id: chain.id,
+              permission: { [Op.in]: ['admin', 'moderator'] },
+            }
+          : {
+              offchain_community_id: community.id,
+              permission: { [Op.in]: ['admin', 'moderator'] },
+            },
+        include: [models.Address],
+        order: [['created_at', 'DESC']],
+      }),
+      // most active users
+      new Promise(async (resolve, reject) => {
+        try {
+          const thirtyDaysAgo = new Date(
+            (new Date() as any) - 1000 * 24 * 60 * 60 * 30
+          );
+          const activeUsers = {};
+          const where = { updated_at: { [Op.gt]: thirtyDaysAgo } };
+          if (community) where['community'] = community.id;
+          else where['chain'] = chain.id;
 
-        // @ts-ignore
-        monthlyComments.concat(monthlyThreads).forEach((post) => {
-          if (!post.Address) return;
-          const addr = post.Address.address;
-          if (activeUsers[addr]) activeUsers[addr]['count'] += 1;
-          else
-            activeUsers[addr] = {
-              info: post.Address,
-              count: 1,
-            };
-        });
-        const mostActiveUsers_ = Object.values(activeUsers).sort((a, b) => {
-          return ((b as any).count - (a as any).count);
-        });
-        resolve(mostActiveUsers_);
-      } catch (e) {
-        reject(new Error('Could not fetch most active users'));
-      }
-    }),
-    models.sequelize.query(`
-     SELECT id, title, stage FROM "OffchainThreads"
-     WHERE ${communityOptions} AND (stage = 'proposal_in_review' OR stage = 'voting')`, {
-        replacements,
-        type: QueryTypes.SELECT,
-      }
-    ),
-  ]));
+          const monthlyComments = await models.OffchainComment.findAll({
+            where,
+            include: [models.Address],
+          });
+          const monthlyThreads = await models.OffchainThread.findAll({
+            where,
+            attributes: { exclude: ['version_history'] },
+            include: [{ model: models.Address, as: 'Address' }],
+          });
 
-  const [threads, comments] = threadsCommentsReactions as any;
+          (monthlyComments as any).concat(monthlyThreads).forEach((post) => {
```

Comment:
Not my code, just my lint, but you get a types complaint if you try to concat different types. I suppose I could make a simple interface like ForumPost or UserPost that just has an Address property, and cast monthlyComments + monthlyThreads as a UserPost, since an Address object is all that's needed to perform the activeUser count?